src/main/java/org/repodriller/domain/Commit.java

```diff
@@ -76,29 +76,32 @@ public Developer getAuthor() {
 	public String getMsg() {
 		return msg;
 	}
-	
+
 	public Developer getCommitter() {
 		return committer;
 	}
 
 	public String getParent() {
 		return parent;
 	}
-	
+
+	/* TODO Deprecate this in favor of the other version. */
 	public void addModification(String oldPath, String newPath, ModificationType change, String diff, String sc) {
```

Comment:
This method creates a `Modification`, because, in my view, it does not make sense to create a `Modification` outside of the context of a `Commit`. So, `Commit` is like the aggregate root of everything. As the construction of `Modification` is quite simple, I don't think it makes such a difference.

We can use your new version, but if the old `addModification` is not used by anyone, then, you can delete it, as this classes are supposed to be only instantiated by our framework (i.e., we are not breaking anyone, so let's keep the codebase as short as possible)

Reply:
Here are my thoughts:

I like the proposed API better because it honors Encapsulation. Since a Modification isn't a private class of a Commit, we should use the Modification type where appropriate.

If a Modification ever becomes more complicated, then this API has to be changed to accommodate the new fields. If the API just accepts a Modification then the Commit code is unaffected.

I also think the name (`addModification`) implies that the argument is a Modification.

If you're OK with the change, I'll just remove the old API. Let me know.

Reply:
Go for it!